### PR TITLE
Fix shipping rate discount factory

### PR DESCRIPTION
**Pull request by mamhoff**
This factory would violate a not-null constraint.

### CODE REVIEW

File: lib/solidus_friendly_promotions/testing_support/factories/friendly_shipping_rate_discount_factory.rb [hunk 1]
```diff
@@ -4,7 +4,11 @@
...
     amount { BigDecimal("-4.00") }
     shipping_rate
-    promotion_action { SolidusFriendlyPromotions::Actions::AdjustShipment.new }
+    promotion_action do
+      promotion = create(:friendly_promotion)
+      hundred_percent = SolidusFriendlyPromotions::Calculators::Percent.new(preferred_percent: 100)
+      SolidusFriendlyPromotions::Actions::AdjustShipment.create!(promotion: promotion, calculator: hundred_percent)
+    end
     label { "10% off" }
   end
```
Comment (davecandlescience) [inline lib/solidus_friendly_promotions/testing_support/factories/friendly_shipping_rate_discount_factory.rb:13]: Nit: the amount, label, and calculator percentage do not match which leads to confusion

Comment (mamhoff) [reply lib/solidus_friendly_promotions/testing_support/factories/friendly_shipping_rate_discount_factory.rb]: Fixed I think

Comment (davecandlescience) [reply lib/solidus_friendly_promotions/testing_support/factories/friendly_shipping_rate_discount_factory.rb]: Mostly... line 9 var name is hundred_percent still

**Review by davecandlescience — APPROVED** [review]
Nice

**Pull request merged**